configure.ac

```diff
@@ -880,6 +886,46 @@ case "$host" in
 esac
 
 
+dnl
+dnl Checking Multipath TCP support on Linux
+dnl
+case "$host" in
+	*-*-linux*)
+		AC_MSG_CHECKING([Multipath TCP support ])
+		AS_IF([test "x$enable_mptcp" != xno],
+				  [AC_RUN_IFELSE( [AC_LANG_PROGRAM([[
+				  		  #include <stdio.h>
+						  #include <stdlib.h>
+						  #include <sys/types.h> 
+						  #include <sys/socket.h>
+						  #ifndef IPPROTO_MPTCP
+						  #define IPPROTO_MPTCP 262
+						  #endif
+						  int x=0;
+						  ]],
+						  [[
+						  int s= socket(AF_INET, SOCK_STREAM, IPPROTO_MPTCP);
+						  if(s!=-1)
+						  {
+						    close(s);
+						    return(0);
+						  }
+						  else
+						  {
+  						    return(-1);
+						  }
+						  ]]
+						  ) ],
```

Comment:
> I don't think this is needed: often these days, the binaries are compiled on a build host, probably with a different kernel that the one end-users will have.
> 
> In other words, I don't think you should check if the kernel support this at build time. It has to be done later.
> 
> What might be useful is to only define `ENABLE_MPTCP` on Linux target. Then you don't need to have `#if defined(TARGET_LINUX) && defined(ENABLE_MPTCP)` in the code, just `ifdef ENABLE_MPTCP`

Reply:
@matttbe sorry I've got no experience with autoconf. Do you mean something like this:

```
dnl
dnl Enable checking Multipath TCP support on Linux
dnl
case "$host" in
	*-*-linux*)
		AC_DEFINE(ENABLE_MPTCP, 1, [Enable checking Multipath TCP support])
	;;
esac
```

Reply:
I don't think you should do that there, but in the default option with `AC_ARG` (BTW, it should be an `ENABLE`, not a `WITH`), something like that (not tested):

```
AC_ARG_ENABLE(
	[mptcp],
	[AS_HELP_STRING([--disable-mptcp], [disable ultipath TCP support @<:@default=yes@:>@ on Linux only])],
	,
	[
		case "$host" in
			*-*-linux*) enable_mptcp="yes";;
			*) enable_mptcp="no" ;;
		esac
	]
)
```